### PR TITLE
Adding first version of new module

### PR DESCRIPTION
**Pull request by danjbh**
## what & why?
* New public module for deploying Amazon Athena to AWS

### CODE REVIEW

**Comment by danjbh** [comment]
Looks like I'm missing support for the data catalog in this module.. I'll get it added ASAP and will reopen this for review.

File: main.tf [hunk 1]
```diff
@@ -1,13 +1,98 @@
...
+
+resource "aws_kms_key" "default" {
+  count = local.enabled && var.create_kms_key ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  deletion_window_in_days = var.athena_kms_key_deletion_window
+  description             = "Athena KMS Key"
```
Comment (korenyoni) [inline main.tf]:
```suggestion
  description             = "Athena KMS Key for ${module.this.id}"
```

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,98 @@
...
+}
```
Comment (korenyoni) [inline main.tf:22]: Missing tags

```suggestion
  tags = module.this.tags
}
```

**Review by korenyoni — CHANGES_REQUESTED** [review]
Couple minor changes requested, LGTM otherwise

**Review by korenyoni — APPROVED** [review]
LGTM

**Pull request merged**